xbmc/utils/GroupUtils.cpp

```diff
@@ -147,11 +154,14 @@ bool GroupUtils::Group(GroupBy groupBy, const std::string &baseDir, const CFileI
   return true;
 }
 
-bool GroupUtils::GroupAndSort(GroupBy groupBy, const std::string &baseDir, const CFileItemList &items, const SortDescription &sortDescription, CFileItemList &groupedItems, GroupAttribute groupAttributes /* = GroupAttributeNone */)
+bool GroupUtils::GroupAndMix(GroupBy groupBy, const std::string &baseDir, const CFileItemList &items, CFileItemList &groupedItemsMixed, GroupAttribute groupAttributes /* = GroupAttributeNone */)
 {
-  if (!Group(groupBy, baseDir, items, groupedItems, groupAttributes))
+  CFileItemList ungroupedItems;
+  if (!Group(groupBy, baseDir, items, groupedItemsMixed, ungroupedItems, groupAttributes))
     return false;
 
-  groupedItems.Sort(sortDescription);
+  // add all the ungrouped items as well
+  groupedItemsMixed.Append(ungroupedItems);
```

Comment:
If you look at the individual commits I first removed the existing `GroupAndSort()` because it wasn't used anywhere. Then I introduced a new `GroupAndMix()` so the overall diff looks a bit odd.